File: EJEMPLOS PATRONES/Python/Facade.py
class BankSystem:
    def pay(self):
        pass


class Email:
    def notifyPay(self):
        pass


class BillingSystem:
    pass


class CustomerRelationshipManager:
    pass


class Facade:
    def __init__(self):
        self._bankSystem = BankSystem()
        self._billingSystem = BillingSystem()
        self._CRM = CustomerRelationshipManager()
        self._Email = Email()

    def pey(self):
        self._bankSystem.pay()
        self._Email.notifyPay()


def main():
    facade = Facade()
    facade.pay()

File: Parte1/EJEMPLOS PATRONES/PHP/Facade.php
<?php

class Facade {
    public function __construct() {
        $this->bankSystem = new BankSistem();
        $this->billingSystem = new BillingSystem();
        $this->CRM = new CustomerRelationshipManager();
        $this->Email = new Email();
    }
    public function pay() {
        $this->BankSistem::pay;
        $this->Email::notify;
    }
}

class BankSistem {
    public function __construct();
    public function pay();
}

class Email {
    public function notify();
}




